### PR TITLE
Car department project dropdown unsetting

### LINKED ISSUE #390: Car, Project, Department
- [x] You should be able to choose Car, Project and Department as 0 - Diverse and choose car.

- [x] Also create migrations for making the ProjectID, CarID and DepartmentID able to be NULL and Default NULL in tables like accountline.

### LINKED ISSUE #403: Picking default car, department and project based from accountplan on bank.tabbankaccount
- [x] On save, try fetching car, department and project from result account plan then hovedboks konto

### LINKED ISSUE #427: Unsetting accountplan on tab bankaccount
When unsetting accountplan on tab.bankaccount, unset Car Project Department

### PR DESCRIPTION
**Pull request by agvald**
Closes #390 
Closes #403 
Closes #427

### CODE REVIEW

**Review by mladjo2505 — DISMISSED** [review]
All looks good, but needs extensive testing before we allow it to go on live. 
Also there are some discussions about what is the wanted behaviour for some pages. I leave that for final review.

**Comment by agvald** [comment]
Before this PR is meged we need to:
 - rebase on master after PR #379 is merged
 - add some small additional changes to journal.edit

EDIT: All done

**Review by mladjo2505 — APPROVED** [review]
All good with new stuff

File: code/lib/db/db_mysqli.class.php [hunk 1]
```diff
@@ -340,12 +344,12 @@ function update($input) {
 
                 foreach($pk_data as $field => $value) {
                     #print "$field = $value<br>\n";
-                  #Should we check if it is functions here? no.
-                    if ($value == DB_NULL_PLACEHOLDER) {
+                    #Should we check if it is funsctions here? no.
```
Comment (stoivo) [inline code/lib/db/db_mysqli.class.php]: functions?

File: modules/salary/view/template.php [hunk 2]
```diff
@@ -209,7 +209,7 @@ function switchActive(id)
             print "<td>";
         if($accountplan->EnableCar == 1)
         {
-            print $_lib['form3']->Car_menu3(array('table'=>$db_table2, 'field'=>'CarID', 'pk'=>$line->SalaryConfLineID, 'value'=>$line->CarID, 'active_reference_date'=>$_lib['sess']->get_session('LoginFormDate')));
+            print $_lib['form2']->Car_menu2(array('table'=>$db_table2, 'field'=>'CarID', 'pk'=>$line->SalaryConfLineID, 'value'=>$line->CarID, 'active_reference_date'=>$_lib['sess']->get_session('LoginFormDate'), 'unset' => true));
```
Comment (stoivo) [inline modules/salary/view/template.php:212]: Check this on @agvald 💻

File: modules/bank/view/record.inc [hunk 3]
```diff
@@ -182,31 +182,47 @@ function do_bank_update($table_name, &$accountline, $Period) {
...
 
         // Allow unsetting a resultaccount
         if (!isset($accountline['ResultAccountPlanID'])) {
-            // set value to "" to not trigger errors like:
-            // get_accountplan_object(): Mangler AccountplanID
             $accountline['ResultAccountPlanID'] = DB_NULL_PLACEHOLDER;
         }
 
-        // Allow unsetting a Car
+        $reskontroaccountplan = $accounting->get_accountplan_object($accountline['ReskontroAccountPlanID']);
```
Comment (stoivo) [inline modules/bank/view/record.inc]: Where is this

File: modules/bank/view/record.inc [hunk 4]
```diff
@@ -182,46 +182,52 @@ function do_bank_update($table_name, &$accountline, $Period) {
 
         // Allow unsetting a reskontro
         if (!isset($accountline['ReskontroAccountPlanID'])) {
+            unset($accountline['CarID']);
```
Comment (stoivo) [inline modules/bank/view/record.inc]: show me

**Review by stoivo — CHANGES_REQUESTED** [review]
Translate

**Review by stoivo — APPROVED** [review]

**Pull request merged**